app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :read, [Post, Comment]
+
+    return unless user.present?
+
+    can %i[create destroy], [Post, Comment], author: user
+
+    return unless user.role?
+
+    can :destroy, [Post, Comment]
+  end
```

Comment:
- [ ] You are not actually taking care of the `admin` user role.
- A user can delete a post if it is theirs or if they have an admin role
- A user can delete a comment if it is theirs or if they have an admin role

It would be great to create the `user.admin?` method inside the user model and use it here😉. Let's fix that